WordPress/src/main/java/org/wordpress/android/models/networkresource/ListState.kt

```diff
@@ -111,15 +116,18 @@ sealed class ListState<T>(val data: List<T>) {
      * It's initialized either from the previous state or from the transformed data using
      * [ListState.transform].
      *
-     * @param errorMessage will be the error string received from the API. It can also be used to show connection errors
-     * where the network is not available.
+     * @param errorMessage will be the error string received from the API.
+     * @param errorMessageResId can be used to propagate error resourceIds from ViewModels to Views.
      */
-    class Error<T> private constructor(data: List<T>, val errorMessage: String?)
-        : ListState<T>(data) {
-        constructor(previous: ListState<T>, errorMessage: String?)
-                : this(previous.data, errorMessage)
+    class Error<T> private constructor(
```

Comment:
I agree! 

I was considering something like [this](https://github.com/wordpress-mobile/WordPress-Android/commit/24027bda4309f09d1a3e6010fb00567dc24b9544), but I didn't like it too much

Reply:
I think that was on the right path, maybe we could bring just a little more structure to it. We should also consider whether the error type should be in the `ListState` base class as well. I really like to enforce people to think about the error handling and that'd help with that. It might potentially make it a bit too verbose though, so that's the other consideration.

I've opened an issue for this and assigned it to the both of us for now https://github.com/wordpress-mobile/WordPress-Android/issues/8744. We can check it out when we get a chance and update its usage everywhere.